File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/codeforces_70_B.java

import java.util.Scanner;

public class codeforces_70_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();
        int ans = 1, sumL = 0;
        String input = scanner.nextLine();
        String[] sentences = input.split("[.?!]");
        for (String s : sentences) {
            s = s.strip() + ".";
            int L = s.length();
            if (L > 1) {
                if (L > n) {
                    System.out.println("Impossible");
                    System.exit(0);
                }
                if (sumL + L + (sumL > 0 ? 1 : 0) > n) {
                    ans++;
                    sumL = L;
                } else {
                    sumL = sumL + L + (sumL > 0 ? 1 : 0);
                }
            }
        }
        System.out.println(ans);
    }
}

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/java/python/codeforces_70_B.py

import sys

n = int(sys.stdin.readline())
input = sys.stdin.readline().strip()
max_len = -1
msg_length = [0] * (len(input) // 2)
count = 0
idx = 0

for i in range(len(input)):
    c = input[i]
    if c in '.?!':
        msg_length[idx] = count + 1
        idx += 1
        if count + 1 > max_len:
            max_len = count + 1
        count = 0
        i += 1
    else:
        count += 1

if max_len > n:
    print("Impossible")
else:
    ans = 0
    i = 0
    while i < idx:
        l = msg_length[i]
        while i < idx - 1 and l + msg_length[i + 1] + 1 <= n:
            l += msg_length[i + 1] + 1
            i += 1
        ans += 1
        i += 1
    print(ans)

